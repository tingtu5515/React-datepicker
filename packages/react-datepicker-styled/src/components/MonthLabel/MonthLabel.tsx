import React from 'react'
import Text from '../Text'

interface MonthLabelProps {
  label: string
}

const MonthLabel = ({label}: MonthLabelProps) => {
  return (
    <Text
      fontFamily="Montserrat"
      fontSize="14px"
      fontWeight={600}
      lineHeight={1.57}
      color="#343132"
    >
      {label}
    </Text>
  )
}

export default MonthLabel
