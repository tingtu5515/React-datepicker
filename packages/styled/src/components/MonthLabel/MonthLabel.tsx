import React from 'react'
import Text from '../Text'
// eslint-disable-next-line import/no-unresolved
import {MonthLabelTheme} from '../../@types/theme'
import useThemeProps from '../../hooks/useThemeProps'
import globalStyles from '../../globalStyles'

interface MonthLabelProps {
  label: string
}

const MonthLabel = ({label}: MonthLabelProps) => {
  const theme: MonthLabelTheme = useThemeProps({
    fontFamily: globalStyles.fontFamily,
    monthLabelColor: globalStyles.colors.darcula,
    monthLabelLineHeight: 1.57,
    monthLabelFontWeight: 600,
    monthLabelFontSize: '14px',
  })

  return (
    <Text
      fontFamily={theme.fontFamily}
      fontSize={theme.monthLabelFontSize}
      fontWeight={theme.monthLabelFontWeight}
      lineHeight={theme.monthLabelLineHeight}
      // @ts-ignore
      color={theme.monthLabelColor}
      data-testid="MonthLabel"
    >
      {label}
    </Text>
  )
}

export default MonthLabel
