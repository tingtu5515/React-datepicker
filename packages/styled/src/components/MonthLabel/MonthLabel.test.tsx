import * as React from 'react'
import {render} from 'react-testing-library'
import MonthLabel from '.'

test('should match snapshot', () => {
  const {container} = render(<MonthLabel label="Test" />)
  expect(container).toMatchSnapshot()
})
