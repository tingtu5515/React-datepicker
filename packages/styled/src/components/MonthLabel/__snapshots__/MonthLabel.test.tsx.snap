// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  font-family: Montserrat,sans-serif;
  font-size: 14px;
  font-weight: 600;
  color: #343132;
  line-height: 1.57;
}

<div>
  <div
    class="c0"
    color="#343132"
    font-family="Montserrat, sans-serif"
    font-size="14px"
    font-weight="600"
  >
    Test
  </div>
</div>
`;
